### ISSUE TITLE
Convert common-code repos to ES6 patterns

### ISSUE THREAD

**Issue by mattpen** [phetsims/tasks#1044]
At 7/30/2020 dev meeting, we decide to create this "chip away" issue to convert common code to modern ES6 patterns.  @ariel-phet asked @pixelzoom to identify patterns and repos to convert.

Patterns:

* Replace `inherit` with `class`.
* Replace `self` with `this` and arrow functions.
* Replace `var` with `const` or `let`.
* Replace `MyClass.someMethod =` with `static someMethod(...)`.
* Replace `MyClass.prototype.someMethod.call( this, ...)` with `super.someMethod(...)`

Number of occurrences of `inherit`:

| repo | start | currently | done |
| :-- | --: | --: | :--: |
| axon | 2 | 0 |  ✓ |
| blast | 5 | 0 | ✓ |
| chains | 1 | 0 |  ✓ |
| dot | 25 | 0 | ✓ |
| griddle | 4 | 0 | ✓ |
| joist | 26 | 0 | ✓ |
| kite | 20 | 0 | ✓ |
| nitroglycerin | 38 | 0 | ✓ |
| phet-core | 10 | 0 | ✓ |
| phetcommon | 2  | 0 | ✓ |
| scenery-phet | 117 | 0 | ✓ |
| scenery | 130 | 0 | ✓ |
| shred | 17 | 0 | ✓ |
| sun | 54 | 0 | ✓ |
| tandem | 1 | 0 | ✓ |
| twixt | 10 | 0 | ✓ |
| vegas | 18 | 0 | ✓ |
| vibe | 2 | 0 | ✓ |

**Comment by ariel-phet** [comment]
@pixelzoom when you make that checklist please make it as the first comment and delete both this one and the initial comment :)

**Comment by samreid** [comment]
A reminder that `class` can extend `inherit` but `inherit cannot extend `class`, so do not convert a base type from `inherit` to `class` unless all of its subtypes have are already using `class`.

**Comment by pixelzoom** [comment]
@ariel-phet would like me to knock off a couple of "low risk" conversions in the near future.

**Comment by pixelzoom** [comment]
I finished 2/2 in vibe and did 4/17 in vegas.

**Comment by pixelzoom** [comment]
@ariel-phet would like to keep this assigned to me, and chip away when I have time/motivation.

**Comment by zepumph** [comment]
@ariel-phet assigned @jonathanolson for dot this week. @Denz1994 will take on shred.

Thanks!

**Comment by samreid** [comment]
I converted a few usages in NumberPicker as part of https://github.com/phetsims/scenery/issues/1078

**Comment by pixelzoom** [comment]
> I converted a few usages in NumberPicker as part of phetsims/scenery#1078

... and apparently didn't mention this issue, so commits related to ES6 conversion (this issue) are not identified. And there are many commits - I count 8. This is consequently going to be time-consuming to review, so I've moved review (and clarification) to phetsims/scenery-phet#626.

**Comment by samreid** [comment]
The 3 commits to convert to ES6 "Convert var=>const" "Convert to class" and "Convert to class" are now all linked to this issue via commit comments (see above).

**Comment by Denz1994** [comment]
From Dev Meeting on 09/24/20:

@pixelzoom mentioned that Joist has some larger classes (ScreenView.js for example) @ariel-phet likes the progression on this issue. Let's keep it up.

**Comment by pixelzoom** [comment]
10/1/2020 dev meeting:

sun is mostly converted. @jonathanolson will convert `MutableOptionsNode` and `Popupable`.

`Panel` is going to be a lot of work, because there are 47 occurrences of `inherit( Panel,` that must be converted first.

**Comment by jbphet** [comment]
Over Slack, @pixelzoom was making a PSA about converting common code classes to ES6, and said "I'm not very familiar with shred, so I verified by fuzz testing sims that use shred. You may want to test drive to make sure there's nothing wrong that fuzz testing didn't catch."  I did a sanity test on Build an Atom and Isotopes and Atomic Mass, and all looks well there.

**Comment by pixelzoom** [comment]
10/15/2020 dev meeting:

* @jonathanolson is going to handle conversion of phet-core
* @jonathanolson is going to handle conversion of scenery-phet FormulaNode (see phetsims/scenery-phet#634)
* before converting scenery and tandem.PhetioObject, large numbers of subclasses in sim-specific code must be converted

**Comment by Denz1994** [comment]
From dev meeting on 11/23/20:

@jonathanolson will add phet-core conversion to his priority list. The remaining repos are still on hold until all sims are published.

**Comment by pixelzoom** [comment]
12/10/2020 dev meeting:

@jonathanolson is handling the remaining items in the list.  Waiting on simulations to be fully converted.

**Comment by jbphet** [comment]
Sims have been converted (see #1051), so we should now be able to move forward on this.

**Comment by jonathanolson** [comment]
Inherit and module-code conversion looks complete!

There is still some ES5-style code in things like scenery/tests (or other code?), leaving on meeting to decide what priority those should be!

**Comment by ariel-phet** [comment]
We feel like this issue can be closed and devs can feel free to convert any vestigial code not covered by this issue to ES6
